front/components/vaults/VaultSideBarMenu.tsx

```diff
@@ -0,0 +1,252 @@
+import {
+  CloudArrowLeftRightIcon,
+  CommandLineIcon,
+  FolderIcon,
+  GlobeAltIcon,
+  Item,
+  LockIcon,
+  PlanetIcon,
+  Tree,
+} from "@dust-tt/sparkle";
+import type {
+  DataSourceOrViewInfo,
+  LightWorkspaceType,
+  VaultType,
+} from "@dust-tt/types";
+import { DATA_SOURCE_OR_VIEW_CATEGORIES } from "@dust-tt/types";
+import { useRouter } from "next/router";
+import type { ReactElement } from "react";
+import { Fragment, useState } from "react";
+
+import { CONNECTOR_CONFIGURATIONS } from "@app/lib/connector_providers";
+import {
+  useVaultDataSourceOrViews,
+  useVaultInfo,
+  useVaults,
+} from "@app/lib/swr";
+
+interface VaultSideBarMenuProps {
+  owner: LightWorkspaceType;
+}
+
+export default function VaultSideBarMenu({ owner }: VaultSideBarMenuProps) {
+  const { vaults, isVaultsLoading } = useVaults({ workspaceId: owner.sId });
+
+  if (!vaults || isVaultsLoading) {
+    return <></>;
+  }
+
+  return (
+    <div className="flex flex-col px-3">
+      <Item.List>
+        {vaults.map((vault) => (
+          <Fragment key={`vault-${vault.sId}`}>
+            <Item.SectionHeader
```

Comment:
Ok I was gonna change the API to return in the right order, but let me do it on the UI side.

Reply:
Fixed.

Reply:
👍 
Still we have one section header per vault, instead of a single "private" section : 
<img width="300" alt="Screenshot 2024-08-12 at 22 00 44" src="https://github.com/user-attachments/assets/b5afdd7e-2d12-4c79-8a93-bec46fc82ca4">

Reply:
Oh I see. Should be fixed.
<img width="318" alt="image" src="https://github.com/user-attachments/assets/b719558c-3b1a-4051-8c41-483d231ae189">